app/models/internal/event.rb

```diff
@@ -106,6 +106,12 @@ def buildings
                        .new.get_teaching_event_buildings
     end
 
+    def type_id=(value)
+      super(value)
+
+      assign_attributes({ is_online: true }) if online_event? && is_online.nil?
```

Comment:
Does this need to use `assign_attributes`? will it not work with the accessor (the `is_online.nil?` is redundant as well IMO - if someone sets an `is_online` to `false` and then changes the type to online event we should just toggle it to `true`:

```
self.is_online = true if online_event?
```

Reply:
Ah, I was missing the `self`